Kernel/Arch/aarch64/PageDirectory.cpp

```diff
@@ -77,33 +77,9 @@ ErrorOr<NonnullLockRefPtr<PageDirectory>> PageDirectory::try_create_for_userspac
         auto& table = *(PageDirectoryPointerTable*)MM.quickmap_page(*directory->m_directory_table);
         for (size_t i = 0; i < sizeof(m_directory_pages) / sizeof(m_directory_pages[0]); i++) {
             if (directory->m_directory_pages[i]) {
-                table.raw[i] = (FlatPtr)directory->m_directory_pages[i]->paddr().as_ptr() | 7;
+                table.raw[i] = (FlatPtr)directory->m_directory_pages[i]->paddr().as_ptr() | PAGE_DESCRIPTOR;
             }
         }
-
-        // 2 ** MAXPHYADDR - 1
-        // Where MAXPHYADDR = physical_address_bit_width
-        u64 max_physical_address = (1ULL << Processor::current().physical_address_bit_width()) - 1;
-
-        // bit 63 = no execute
-        // bit 7 = page size
-        // bit 5 = accessed
-        // bit 4 = cache disable
-        // bit 3 = write through
-        // bit 2 = user/supervisor
-        // bit 1 = read/write
-        // bit 0 = present
-        constexpr u64 pdpte_bit_flags = 0x80000000000000BF;
-
-        // This is to notify us of bugs where we're:
-        // 1. Going over what the processor is capable of.
-        // 2. Writing into the reserved bits (51:MAXPHYADDR), where doing so throws a GPF
-        //    when writing out the PDPT pointer to CR3.
-        // The reason we're not checking the page directory's physical address directly is because
-        // we're checking for sign extension when putting it into a PDPTE. See issue #4584.
-        for (auto table_entry : table.raw)
-            VERIFY((table_entry & ~pdpte_bit_flags) <= max_physical_address);
```

Comment:
Should this move to somewhere else? Oh I see, it got added somewhere else in the previous commit.

A bit weird to add this code elsewhere in one commit and then delete it in a different commit. Should probably both happen in the same commit?

Reply:
Right, that would be better indeed. I squashed this commit with the previous one, and updated the commit message to include these changes.